Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm

```diff
@@ -645,6 +645,42 @@ static void logProcessPoolState(const WebProcessPool& pool)
 {
     m_weakObserver = adoptNS([[WKProcessPoolWeakObserver alloc] initWithWeakPtr:*this]);
 
+#if ENABLE(NOTIFYD_BLOCKING_IN_WEBCONTENT)
+    const char* messages[] = { "com.apple.WebKit.LibraryPathDiagnostics",
+                                "com.apple.WebKit.dumpUntrackedMallocs",
+                                "org.WebKit.memoryWarning",
+                                "org.WebKit.memoryWarning.begin",
+                                "org.WebKit.memoryWarning.end",
+                                "org.WebKit.lowMemory",
+                                "org.WebKit.lowMemory.begin",
+                                "org.WebKit.lowMemory.end",
+                                "com.apple.WebKit.fullGC",
+                                "com.apple.WebKit.deleteAllCode",
+                                "com.apple.WebKit.dumpGCHeap",
+                                "com.apple.WebKit.showBackForwardCache",
+                                "com.apple.WebKit.showMemoryCache",
+                                "com.apple.WebKit.showRenderTree",
+                                "com.apple.WebKit.showLayerTree",
+                                "com.apple.WebKit.showGraphicsLayerTree",
+                                "com.apple.WebKit.showPaintOrderTree",
+                                "com.apple.WebKit.showLayoutTree",
+                                "com.apple.WebKit.showAllDocuments",
+                                "com.apple.WebKit.logMemStats",
+                                "com.apple.WebKit.logPageState" };
+    m_notifyTokens.reserveInitialCapacity(std::size(messages));
+    for (auto* message : messages) {
+        int notifyToken;
+        notify_register_dispatch(message, &notifyToken, dispatch_get_main_queue(), ^(int token) {
+            for (auto& process : m_processes) {
+                if (!process->canSendMessage())
+                    continue;
+                process->send(Messages::WebProcess::PostNotification(message), 0);
```

Comment:
This repeatedly constructs a WTF::String from a `const char*` for every process.

Reply:
The following would be more efficient:
```
if (!m_processes.isEmpty()) {
    String messageString(message);
    for (auto& process : m_processes)
        process->send(Messages::WebProcess::PostNotification(messageString), 0);
}
```